tests/unit-tests/simple.cpp

```diff
@@ -24,3 +24,33 @@ TEST_CASE("global constant variable can be printed") {
   CHECK(!ss.str().empty());
 }
 
+TEST_CASE("Can create and print line numbers")
+{
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module m{lexicon};
+  impl::Interface_unit unit{lexicon, m};
+
+  impl::Scope* global_scope = unit.global_scope();
+
+  const Name* name = lexicon.make_identifier("bufsz");
+  auto& type = lexicon.get_qualified(Type_qualifier::Const, lexicon.int_type());
+  impl::Var* var = global_scope->make_var(*name, type);
+  var->init = lexicon.make_literal(lexicon.int_type(), "1024");
+
+  auto file_index = lexicon.make_fileindex(lexicon.get_string("hello.cpp"));
```

Comment:
Do we have a good bar for what the `Lexicon` should contain? I know it can hold arbitrary strings (theoretically to represent strings in the user's program), but it feels strange to me to put the filename in the lexicon.

Reply:
I am not sure I understand the question.

Right now we have an 

```
   struct Source_location : Basic_location {
      File_index file = { };    // ID of the file
   };
```

but no mechanism to manufacture the file index. There were functions to manufacture it without an implementation. This patch adds the implementation.

Are you asking whether we need to store the filename string at all?
If so, I believe the answer is yes, since, we want the consumer of the IPR being able to map the nodes to the source locations.

P.S.

Yes. I know that we will have an improved location information in the future, but, for now, I want to make what we already have in IPR work

Reply:
My concern was primarily about what/who will own the filename string. It seems that the `Lexicon` is the owner in this change. I was wondering if the `Lexicon` should definitely be the owner or if it was convenient to put the string in the `Lexicon` for the short-term.

Reply:
Lexicon owns interning and unifications of all the strings, so that any ipr::Strings can be compare with a simple node pointer comparisons. Since filename is presented as a string in `to_filename` API. I think it makes sense to keep the strings under the ownership of the lexicon. Why make an exception for filename strings?

Reply:
Just double-checking. I'm not clear on whether there's supposed to be a `Lexicon` per translation-unit, per file, per project, etc. You'd lose the pointer-comparison properties, for example, if you had a `Lexicon` per file/translation-unit. I don't think `Lexicon` actually takes a stance on this (as it shouldn't), so I guess it's not a valuable question at this point.

Reply:
@Nostacodus has a point here: The primary purpose of `Lexicon` is to server as the set of the type and value parameters to the abstract machine -- not "this is where you put a factory function that didn't exist anywhere else".  Of course, an implementation can inherit from `Lexicon` and adds bunch of functions unrelated to the original purpose.   Made an outcome of this conversation is further elaboration of the purpose of `Lexicon`.

I would think it makes sense for an implementation to have its own structures for keeping tab on its vagaries, e.g. MSVC has its own set of "in-house" functions -- in the case of filename for example, it would take care of path normalization, case-insensitive comparison, UCN, etc.  Those don't need to be part of the IPR repo though.